### PR TITLE
Update README after introducing upstream-extra

### PR DESCRIPTION
**Pull request by lindig**
Signed-off-by: Christian Lindig <christian.lindig@citrix.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,24 +13,24 @@ remote Opam repository:
 
 ## Layout of This Repository
 
-Packages are organised into namespaces. They are equivalent from the
-perspective of the [Opam] user.
+Packages are organised into namespaces:
 
 * `upstream`: packages that we don't control with fixed versions.
+* `upstream-extra`: packages that we don't control with fixed versions.
```
Comment (mseri) [inline README.md]: I would add here that we don't need them as build dependencies. What do you think?

File: README.md [hunk 2]
```diff
@@ -13,24 +13,24 @@ remote Opam repository:
...
 * `xs-extra`: packages that we control, following their respective
   master branch.
 
+The namespaces are treated differently as part of the build process of
+XenServer components but this should not concern an Opam user - packages
+in all namespaces together form the repository and there is no
+difference between them.
+
     ```
-    ./packages
```
Comment (gaborigloi) [inline README.md]: For some reason this block is not rendered correctly, the backticks are visible in the output: https://github.com/lindig/xs-opam/

**Comment by lindig** [comment]
I had mixed-in my blacklist mechanism - took it out.

File: README.md [hunk 3]
```diff
@@ -42,6 +43,20 @@ remote Opam repository. A typical entry consists of three files:
...
+
+    packages/xs/cdrom.0.9.2
+    packages/xs/crc.1.0.0
+    packages/xs-extra/libvirt.djs
+    packages/xs-extra/message-switch.master
+    packages/xs-extra/rrddump.master
+    packages/xs-extra/vncproxy.master
+    packages/xs-extra/wsproxy.master
+    packages/xs/fd-send-recv.1.0.2
+    packages/xs/nbd.2.1.2
```
Comment (gaborigloi) [inline README.md]: Maybe the `xs/` block could be before all the `xs-extra/` ones to make the message of this paragraph more obvious?
```
packages/xs/cdrom.0.9.2
packages/xs/crc.1.0.0
packages/xs/fd-send-recv.1.0.2
packages/xs/nbd.2.1.2
packages/xs/netlink.0.2.1
packages/xs-extra/libvirt.djs
packages/xs-extra/message-switch.master
packages/xs-extra/rrddump.master
packages/xs-extra/vncproxy.master
packages/xs-extra/wsproxy.master
```

**Pull request merged**